### PR TITLE
CURATOR-588  Upgrade ZooKeeper to 3.6.3

### PR DESCRIPTION
**Pull request by eolivelli**
It looks like upgrading to ZK 3.6.3 is not that easy as for 3.6.2

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="UTF-8"?><!--
+e?xml version="1.0" encoding="UTF-8"?><!--
```
Comment (madrob) [inline pom.xml:1]: ?

**Comment by virajjasani** [comment]
Thanks @eolivelli.

Build failure:
```
The command "eval mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V " failed
```
Is this relevant or some other known issue?

File: pom.xml (hunk 1 above)
Comment (eolivelli) [reply pom.xml]: it was not intended, thanks @madrob

**Comment by eolivelli** [comment]
This patch is still a draft, because some tests are still failing

**Comment by eolivelli** [comment]
@cammckenzie I have fixed the problem for zk35 compatibility test.

The weird behaviour is about a ConnectionLoss vs SessionExpired that happens in a test from 3.6.3 onwards

File: curator-framework/src/test/java/org/apache/curator/framework/imps/TestFrameworkEdges.java [hunk 2]
```diff
@@ -670,9 +670,13 @@ public void testFailure() throws Exception
             client.checkExists().forPath("/hey");
             fail();
         }
+        catch ( KeeperException.SessionExpiredException e )
+        {
+            // correct, this happens on ZK 3.6.3+
```
Comment (tisonkun) [inline curator-framework/src/test/java/org/apache/curator/framework/imps/TestFrameworkEdges.java:675]: Why? Shall we setup a link to the reason?

It is a bit weird a patch version bump cause behavior changes.

**Review by tisonkun — COMMENTED** [review]
Thanks for starting this upgrade! Comment inline.

File: curator-framework/src/test/java/org/apache/curator/framework/imps/TestFrameworkEdges.java (hunk 2 above)
Comment (eolivelli) [reply curator-framework/src/test/java/org/apache/curator/framework/imps/TestFrameworkEdges.java:675]: I really do not know why this is happening.
I have looked thru ZK 3.6.3 release notes and I haven't found anything that may be releated.

Comment (tisonkun) [reply curator-framework/src/test/java/org/apache/curator/framework/imps/TestFrameworkEdges.java:675]: Fine. It is not a blocker for client library such as Curator, but is interesting...

Comment (eolivelli) [reply curator-framework/src/test/java/org/apache/curator/framework/imps/TestFrameworkEdges.java:675]: probably Session Expiration is happening earlier with 3.6.3
with 3.6.2 we still see a ConnectionLoss, that usually becomes a SessionExpired when the client is able to connect again and the ZK server already expired it on the server side

**Review by cammckenzie — APPROVED** [review]
Looks good to me, all tests pass.

**Pull request merged**